packages/doenetml-worker-rust/lib-doenetml-core/src/core/document_model/dependency_creation.rs

```diff
@@ -76,10 +73,36 @@ impl DocumentModel {
         // to pass on to the caller
         let mut linked_nodes = Vec::new();
 
+        let mut fn_add_edges = |edges_to_add: Vec<(GraphNode, GraphNode)>| {
+            let mut dependency_graph = self.dependency_graph.borrow_mut();
+            for (from, to) in edges_to_add {
+                // Small sanity check to make sure we don't produce  self loop.
+                // `prop_node` should not ask for a query that results in itself.
+                assert_ne!(
+                    prop_node, to,
+                    "Self-loop detected; DataQuery requested a prop that is the same as the origin prop."
+                );
+                dependency_graph.add_edge(from, to);
+                linked_nodes.push(to);
+            }
+        };
+
+        // Resolve a `PropComponent` to a component index.
```

Comment:
Prop component is now prop source.

Reply:
Could you change the `PropComponent` comment and the variables `prop_component` and `resolve_prop_component` to use "source"?